src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AttendanceService.java

```diff
@@ -77,35 +56,64 @@ public void updateAttendance(long id, UpdateAttendanceDto attendanceDto) throws
                         .createDateTime(attendanceData.getCreateDateTime())
                         .caseNoteId(attendanceData.getCaseNoteId())
                         .build())
-                  .collect(Collectors.toSet());
+                .collect(Collectors.toSet());
     }
 
     public AbsentReasonsDto getAbsenceReasons() {
         return new AbsentReasonsDto(AbsentReason.getPaidReasons(), AbsentReason.getUnpaidReasons());
     }
 
+    @Transactional
+    public void createAttendance(final CreateAttendanceDto attendanceDto) {
 
-    private void applyAttendanceWorkflow(String offenderNo, Attendance attendance) {
-         final var eventOutcome = nomisEventOutcomeMapper.getEventOutcome(
-                 attendance.getAbsentReason(),
-                 attendance.isAttended(),
-                 attendance.isPaid());
+        var attendance = attendanceRepository.save(toAttendance(attendanceDto));
+        entityManager.flush();
+        applyAttendanceWorkflow(attendance);
+    }
 
-         nomisService.putAttendance(offenderNo, attendance.getEventId(), eventOutcome);
+    @Transactional
+    public void updateAttendance(long id, UpdateAttendanceDto attendanceDto) throws AttendanceNotFound {
 
-         if (attendance.getAbsentReason() != null && AbsentReason.getIepTriggers().contains(attendance.getAbsentReason())) {
-             final var caseNote = nomisService.postCaseNote(
-                     attendance.getOffenderBookingId(),
-                     "NEG",//"Negative Behaviour"
-                     "IEP_WARN", //"IEP Warning",
-                     attendance.getComments(),
-                     LocalDateTime.now());
+        final var attendance = attendanceRepository.findById(id)
+                .orElseThrow(AttendanceNotFound::new);
 
-             attendance.setCaseNoteId(caseNote.getCaseNoteId());
-         }
-     }
+        attendance.setAttended(attendanceDto.getAttended());
+        attendance.setPaid(attendanceDto.getPaid());
+        attendance.setAbsentReason(attendanceDto.getAbsentReason());
+        attendance.setComments(attendanceDto.getComments());
+
+        applyAttendanceWorkflow(attendance);
+
+        attendanceRepository.save(attendance);
+    }
+
+
+    private void applyAttendanceWorkflow(Attendance attendance) {
```

Comment:
missing final... :-D 
maybe something like `publishXXX`/`publishToNomis` might be a bit more descriptive? 
(If that's what this is doing)

Reply:
I like it!